File: 프로그래머스/Lv.2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    ans = 0
    max_tangerine = max(tangerine)
    size = [0] * (max_tangerine + 1)
    for i in tangerine:
        size[i-1] += 1
    size.sort(reverse=True)
    for j in size:
        if k <= 0:
            break
        k -= j
        ans += 1
    return ans

File: 프로그래머스/Lv.2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var ans = 0;
    var size = new Array(Math.max.apply(null, tangerine)).fill(0);
    for (let i = 0 ; i<tangerine.length;i++){
        size[tangerine[i]-1] +=1;
    }
    size.sort(function compare(a,b){
        return b-a;
    })
    for(let j = 0; j < size.length;j++){
        if(k<=0){
            break;
        } 
        k-= size[j];
        ans++;
    }

    return ans;
}